wgsl/index.bs

```diff
@@ -10040,16 +10367,31 @@ Note: Refer to [[#func-var-value-analysis]] for the definition of *Vout*(*call*)
 Most built-in functions have tags of:
 - A [=call site tag=] of [=CallSiteNoRestriction=].
 - A [=function tag=] of [=NoRestriction=].
-- For each parameter, a [=parameter tag|tag=] of [=ParameterRequiredToBeUniformForReturnValue=].
+- For each parameter, a [=parameter tag=] of [=ParameterRequiredToBeUniformForReturnValue=].
 
 Here is the list of exceptions:
-- All functions in [[#sync-builtin-functions]] have a [=call site tag=] of  [=CallSiteRequiredToBeUniform=].
-    - The parameter `p` in [[#workgroupUniformLoad-builtin|workgroupUniformLoad]]
-        has a [=parameter tag=] of [=ParameterRequiredToBeUniform=].
-- All functions in [[#derivative-builtin-functions]], [[#texturesample]], [[#texturesamplebias]], and [[#texturesamplecompare]] have a [=call site tag=] of [=CallSiteRequiredToBeUniform=] and a [=function tag=] of [=ReturnValueMayBeNonUniform=].
-- [[#arrayLength-builtin|arrayLength]] has a [=call site tag=] of
-    [=CallSiteNoRestriction=], a [=function tag=] of [=NoRestriction=] and
-    the input parameter `p` has a [=parameter tag=] of [=ParameterNoRestriction=]
+- A call to a function in [[#sync-builtin-functions]]:
+    - Has a [=function tag=] of [=NoRestriction=].
+    - Has a [=call site tag=] of [=CallSiteRequiredToBeUniform.S|CallSiteRequiredToBeUniform.error=].
+        The tag's associated [=witness=] is the source location of the call site, combined with an unnamed [=diagnostic/triggering rule=].
+        - Note: The triggering rule has no name, and so it cannot be filtered.
+    - Additionally for the case of a call to [[#workgroupUniformLoad-builtin|workgroupUniformLoad]],
+        the parameter `p` has a [=parameter tag=] of [=ParameterRequiredToBeUniform.S|ParameterRequiredToBeUniform.error=].
+        The tag's associated [=witness=] is the source location of the call site, combined with an unnamed [=diagnostic/triggering rule=].
+- A call to a function in
+    [[#derivative-builtin-functions]], [[#texturesample]], [[#texturesamplebias]], and [[#texturesamplecompare]]:
+    - Has a [=function tag=] of [=ReturnValueMayBeNonUniform=].
+    - Has a [=call site tag=] of [=CallSiteRequiredToBeUniform.S=]:
+        - Where *S* is:
+            - the NS (new severity) parameter of the [=nearest enclosing diagnostic filter=]
+                for the combination of the call site and the [=trigger/derivative_uniformity=] triggering rule.
+            - [=severity/error=], if no such filter exists.
+        - The tag's associated [=witness=] is the source location of the call site, combined with the [=trigger/derivative_uniformity=] triggering rule.
+
+- [[#arrayLength-builtin|arrayLength]]:
+    - Has a [=function tag=] of [=NoRestriction=].
+    - Has a [=call site tag=] of [=CallSiteNoRestriction=].
+    - The input parameter `p` has a [=parameter tag=] of [=ParameterNoRestriction=]
```

Comment:
I don't think the special case for `arrrayLength` is needed at all any more. It's fine for its parameter tag to be `ParameterRequiredToBeUniformForReturnValue` as per the defaults above (this is what Tint does) and this is future proof for a world where we have variable pointers.